stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1017,11 +1017,33 @@ function rmul!(A::SparseMatrixCSC, b::Number)
     rmul!(A.nzval, b)
     return A
 end
+
 function lmul!(b::Number, A::SparseMatrixCSC)
     lmul!(b, A.nzval)
     return A
 end
 
+function rmul!(A::SparseMatrixCSC, D::Diagonal{T}) where T
```

Comment:
`T` is never used in the function body so I think you can make this signature

`function rmul!(A::SparseMatrixCSC, D::Diagonal)`. (same for lmul)